src/katgpucbf/xbgpu/engine.py

```diff
@@ -433,17 +441,17 @@ def __init__(
         self.dump_interval_s: float = self.timestamp_increment_per_accumulation / adc_sample_rate_hz
 
         self.send_stream = XSend(
-            n_ants=self.n_ants,
-            n_channels=self.n_channels_total,
-            n_channels_per_stream=self.n_channels_per_stream,
+            n_ants=n_ants,
+            n_channels=n_channels_total,
+            n_channels_per_stream=n_channels_per_stream,
             dump_interval_s=self.dump_interval_s,
-            send_rate_factor=self.send_rate_factor,
+            send_rate_factor=send_rate_factor,
             channel_offset=self.channel_offset_value,  # Arbitrary for now - depends on F-Engine stream
             context=self.context,
             packet_payload=dst_packet_payload,
             stream_factory=lambda stream_config, buffers: make_stream(
-                dest_ip=dst.host,
-                dest_port=dst.port,
+                dest_ip=outputs[0].dst.host,
+                dest_port=outputs[0].dst.port,
```

Comment:
@bmerry do you think it is worth protecting this somehow?

Currently `outputs` is created with the correlator output first, but that could change and this could be easy to miss. Something like
```
assert isinstance(outputs[0], XOutput)
``` 
just before the `XSend()` call.

Reply:
I wouldn't worry at this point. This code will all go through major refactoring anyway.